hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1055,6 +1063,68 @@ public void close() throws Exception {
    */
   protected abstract void commit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap);
 
+  /**
+   * Converts the input records to the input format expected by the write client.
+   * @param records records to be converted
+   * @return converted records
+   */
+  protected abstract I convertRecordsToWriteClientInput(HoodieData<HoodieRecord> records);
```

Comment:
may be: `convertHoodieDataToEngineSpecificInput`

Reply:
updating to use this name